cli/command/config/ls.go

```diff
@@ -32,6 +32,9 @@ func newConfigListCommand(dockerCli command.Cli) *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			return RunConfigList(dockerCli, listOpts)
 		},
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			return nil, cobra.ShellCompDirectiveError
+		},
```

Comment:
nit: perhaps we should consider having a `completion.NoComplete()` function, so that we can just reuse that for all of these;

```suggestion
		ValidArgsFunction: completion.NoComplete,
```